ExtractorUtils/BaseExtractor.cs

```diff
@@ -259,29 +257,42 @@ protected virtual Task OnStop()
         /// <param name="disposing">True if disposing</param>
         protected virtual void Dispose(bool disposing)
         {
-            if (!disposedValue)
+            if (disposing)
             {
-                if (disposing)
+                try
+                {
+                    // Cannot be allowed to fail here
+                    Scheduler.ExitAllAndWait().Wait();
+                } catch { }
+                Scheduler.Dispose();
+                EventUploadQueue?.Dispose();
+                TSUploadQueue?.Dispose();
+                foreach (var queue in RawUploadQueues.Values)
                 {
-                    try
-                    {
-                        // Cannot be allowed to fail here
-                        Scheduler.ExitAllAndWait().Wait();
-                    } catch { }
-                    Scheduler.Dispose();
-                    EventUploadQueue?.Dispose();
-                    TSUploadQueue?.Dispose();
-                    foreach (var queue in RawUploadQueues.Values)
-                    {
-                        queue.Dispose();
-                    }
-                    RawUploadQueues.Clear();
-                    Source.Cancel();
-                    Source.Dispose();
+                    queue.Dispose();
                 }
+                RawUploadQueues.Clear();
+                Source.Cancel();
+                Source.Dispose();
+            }
+        }
 
-                disposedValue = true;
+        /// <summary>
+        /// Internal method to dispose asynchronously. Preferred.
```

Comment:
Preferred over what alternative, and is that referencing this function as the better choice?

Reply:
Ah, sure. Yeah, preferred over synchronous dispose. I'll add it where relevant.